aten/src/ATen/cuda/CUDAContext.cpp

```diff
@@ -46,8 +46,17 @@ cudaDeviceProp* getCurrentDeviceProperties() {
 
 cudaDeviceProp* getDeviceProperties(c10::DeviceIndex device) {
   c10::call_once(init_flag, initCUDAContextVectors);
-  if (device == -1) device = c10::cuda::current_device();
-  AT_ASSERT(device >= 0 && device < num_gpus, "device=", device, ", num_gpus=", num_gpus);
+  if (device == -1) {
+    device = c10::cuda::current_device();
+  }
+  TORCH_CHECK(
+    device >= 0 && device < num_gpus,
+    "Could not getDeviceProperties of device ", static_cast<int>(device),
+    " as there are only ", static_cast<int>(num_gpus), " GPUs on this system.  "
+    "This could potentially because the environment variable CUDA_VISIBLE_DEVICES "
```

Comment:
```suggestion
    " as there are only ", static_cast<int>(num_gpus), " GPUs on this system.  "
    "This error often happens when the provided device index is wrong.  "
    "In rare cases, an index that should be correct based on torch.cuda.device_count() "
    "might be wrong because the environment variable CUDA_VISIBLE_DEVICES "
```